### PR TITLE
Remove spotio node pool from e2e [1/2]

### PR DESCRIPTION
**Pull request by mikkeloscar**
Prepare for dropping all spot.io support (#4392)

First we need to drop the node pool created as part of e2e.

### CODE REVIEW

File: cluster/manifests/e2e-resources/pool-reserve.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {{ if eq .Environment "e2e" }}
-{{ range $pool := split "default-worker-splitaz,default-worker,worker-limit-az,worker-instance-storage,default-worker-spotio,worker-node-tests" "," }}
+{{ range $pool := split "default-worker-splitaz,default-worker,worker-limit-az,worker-instance-storage,worker-node-tests" "," }}
```
Comment (aermakov-zalando) [inline cluster/manifests/e2e-resources/pool-reserve.yaml:2]: I think you'll need a deletion as well, otherwise the update would fail.

**Comment by mikkeloscar** [comment]
:+1:

File: cluster/manifests/e2e-resources/pool-reserve.yaml (hunk 1 above)
Comment (mikkeloscar) [reply cluster/manifests/e2e-resources/pool-reserve.yaml:2]: Thanks! I added this and it's working now!

**Comment by aermakov-zalando** [comment]
👍

**Pull request merged**